dotenv.cabal

```diff
@@ -84,10 +84,10 @@ executable dotenv
 library
   exposed-modules:      Configuration.Dotenv
                       , Configuration.Dotenv.Environment
-  other-modules:        Configuration.Dotenv.Parse
-                      , Configuration.Dotenv.ParsedVariable
                       , Configuration.Dotenv.Text
-                      , Configuration.Dotenv.Types
+                      , Configuration.Dotenv.ParsedVariable
```

Comment:
@CristhianMotoche I would recommend using a `Internal` module that re-exports internal details rather than directly exposing them for a few reasons:

- An internal module makes it clear to consumers that its API is more subject to change than the public API.
- Hiding all internal details behind a single module allows for greater refactoring flexibility; for example, you could change the name of the ParsedVariable or split its content without consumers noticing.

Reply:
Thanks for the suggestion @sestrella ! I updated this PR. Please, let me know if this new module structure look good. We'll need to do a major version release, though, since I stopped exporting `Configuration.Dotenv.Environment` in favor of `Configuration.Dotenv.Internal`.